polaris-patterns/src/components/PolarisPatternsProvider/PolarisPatternsProvider.tsx

```diff
@@ -0,0 +1,57 @@
+import type {PropsWithChildren} from 'react';
+import React, {createContext, useEffect} from 'react';
+import {I18nextProvider, initReactI18next} from 'react-i18next';
+import {createInstance} from 'i18next';
+
+import {DEFAULT_I18N_DETAILS} from '../../configure';
+
+export const I18nDetails = createContext(DEFAULT_I18N_DETAILS);
+
+const i18next = createInstance({
+  fallbackLng: DEFAULT_I18N_DETAILS.locale,
+  interpolation: {
+    escapeValue: false,
+  },
+  resources: {},
+  returnNull: false,
+});
+
+i18next.use(initReactI18next).init();
+
+interface TranslationDictionary {
+  [key: string]: string | TranslationDictionary;
+}
+
+interface I18nDetails {
+  locale: string;
+  country?: string;
+  currency?: string;
+  timezone?: string;
```

Comment:
Perhaps we can call this `ianaTimezone` to be explicit (and to match the [`Shop.ianaTimezone` field](https://shopify.dev/docs/api/admin-graphql/2022-07/objects/Shop#field-shop-ianatimezone))

```suggestion
  ianaTimezone?: string;
```

Reply:
Since you raised the question of keeping consistent with the `Intl` library, I'm not sure now if this should be 
 `timeZone` to match the field used in [`Intl.DateTimeFormat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/DateTimeFormat).